R/dash.R

```diff
@@ -292,28 +319,79 @@ Dash <- R6::R6Class(
 
         dep_list <- c(private$dependencies_internal,
                       private$dependencies,
-                      private$dependencies_user)       
- 
-        dep_pkg <- get_package_mapping(filename, 
+                      private$dependencies_user)
+
+        dep_pkg <- get_package_mapping(filename,
                                        keys$package_name,
-                                       clean_dependencies(dep_list) 
+                                       clean_dependencies(dep_list)
                                        )
 
-        dep_path <- system.file(dep_pkg$rpkg_path, 
+        dep_path <- system.file(dep_pkg$rpkg_path,
                                 package = dep_pkg$rpkg_name)
 
         response$body <- readLines(dep_path,
-                                   warn = FALSE, 
+                                   warn = FALSE,
                                    encoding = "UTF-8")
         response$status <- 200L
-        response$set_header('Cache-Control', 
-                            sprintf('public, max-age=%s', 
+        response$set_header('Cache-Control',
+                            sprintf('public, max-age=%s',
                                     components_cache_max_age)
                             )
         response$type <- get_mimetype(filename)
         TRUE
       })
 
+      dash_assets <- paste0(self$config$routes_pathname_prefix, private$assets_folder, "/*")
```

Comment:
shouldn't this one be `assets_url_path` instead of `assets_folder`? Perhaps with its leading `/` stripped...

https://github.com/plotly/dash/blob/69b112df8153ad416fb279944e686cec7dc524a1/dash/dash.py#L158-L161

Reply:
I suspect that the end result might be the same, but I agree with you, logically I should use the URL here. Will fix.

Reply:
They need not be the same though... we might have:
`routes_pathname_prefix='my/app'`
`assets_url_path='/assets'`
`assets_folder='some/local/path'`
(Incidentally I don't know why we include a leading slash in `assets_url_path` if we're just going to strip it... dash does it this way too so not really a question for you, I just don't understand it)

Then suppose we have a file `some/local/path/subfolder/pink.css`, we want that to be served at a URL like `http://localhost:8080/my/appassets/subfolder/pink.css?123456`

I'd think then that the easiest way to match these up would be (while walking the assets folder) to make a mapping of the url full path (stripping off the host and query string) to the file local path, in this case `'my/appassets/subfolder/pink.css' -> 'some/local/path/subfolder/pink.css'`

Reply:
Ah, you're right - good point. I'll ensure that the URL path and assets folder can be as distinct as needed. I've already written regex to parse and excise the host information and `?...` bits, so that part should be :cake:.

Reply:
fixed in fd821fd4bcef02e2bba25afe8708cef55cbdc2d1